File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    def decode(number):
        result = [0 for _ in range(n)]
        index = -1
        while number > 0:
            d = number % 2
            if d == 1:
                result[index] = d
            number = number // 2
            index -= 1
        return result
    
    for i in range(n):
        x = decode(arr1[i])
        y = decode(arr2[i])
        arr = ""
        for i, j in zip(x, y):
            if i == 0 and j == 0:
                arr += " "
            else:
                arr += "#"
        answer.append(arr)
        
    
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = []
    
    for (let i = 0; i < n; i++) {
        const digit = (arr1[i] | arr2[i]).toString(2)
        const result = [...digit].map((d) => d === "1" ? "#" : " ").join("")
        answer.push(result)
    }

    return answer.map(el => " ".repeat(n - el.length) + el);
}